design/one-pager-default-resource-class.md

```diff
@@ -90,6 +90,14 @@ spec:
 
 Internally, Crossplane will first check to see if a resource class is referenced. If not, it will check to see if a class annotated as `default` has been created for the given kind. Ultimately, if one does not exist, it will fail to provision the resource.
 
+## Controllers
+
+Currently, each Crossplane resource kind (i.e. GKE Cluster, AWS S3 Bucket, etc.) has a controller that reconciles claims for that resource by binding them to the corresponding managed type. These controllers use [predicates](https://github.com/negz/crossplane/blob/resourceful/pkg/resource/predicates.go)(TEMPORARY LINK) to ensure that there is a provisioner defined for the class referenced by the claim. If the claim contains no reference to a class, the controller will not act on the claim.
+
+Default resource classes require an additional controller for each resource kind that watches for claims of that resource that have no class reference defined. The controller will check for this using predicates in the same fashion as the claim controller. Upon discovery of a claim without a class reference, the controller searches for a class with an `abstractResource` field that matches the claim kind, and a `default` field set to `true`.
```

Comment:
I suspect we could design this as one controller that watches for all known resource claim kinds without a resource class. Even with concrete resource classes (i.e. many kinds of resource class) we could likely implement this assuming resource classes satisfied an interface like:

```go
type Provisioner struct {
    Name    string
    Default bool
}

type Provisionable interface {
    GetProvisioners() []Provisioner
    SetProvisioners(p []Provisioner)
}

type Class interface {
    runtime.Object
    Provisionable
    // ...Other embedded interfaces
}

Reply:
This is much cleaner and reduced the implementation burden on anyone creating a new resource type. Thanks for the feedback :)